core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -475,7 +475,52 @@ class DAGSchedulerSuite
     assert(results === Map(0 -> 42, 1 -> 43))
     assertDataStructuresEmpty()
   }
+  
+  test("Test taskAbort after multiple stage failures.") {
```

Comment:
Given the (potential) bug I mentioned above, can you add a test that makes sure that if many tasks in a stage fail with FetchFailed messages (say, 10), the map/reduce stages are still retried correctly?